core/src/main/java/org/apache/calcite/util/Util.java

```diff
@@ -1048,6 +1048,22 @@ public static String getStackTrace(Throwable t) {
     return sw.toString();
   }
 
+  /**
+   * Ignores the RuntimeException if it is an AccessControlException otherwise throws.
+   *
+   * <p>Checks a RuntimeException for AccessControlException without importing JDK
+   * classes that are deprecated in <a href="https://openjdk.org/jeps/411">JDK 17</a>
+   * with the anticipated removal of Java security manager.
+   *
+   * @param e RuntimeException to check if it is an AccessControlException
+   */
+  @API(since = "1.33", status = API.Status.EXPERIMENTAL)
+  public static void ignoreAccessControlException(RuntimeException e) {
+    if (!"java.security.AccessControlException".equals(e.getClass().getName())) {
+      throw e;
+    }
+  }
```

Comment:
Do we really need this hacky check? `AccessControlException` is a subclass of `SecurityException`. The latter is not deprecated (https://download.java.net/java/early_access/jdk21/docs/api/java.base/java/lang/SecurityException.html) so I assume we can do regular `catch(SecurityException se)` where needed can't we?

Reply:
Seems like a reasonable idea. From https://openjdk.org/jeps/411

```
...
We will not deprecate some classes in the java.security package that are related to the Security Manager, for various reasons:
...
SecurityException — A runtime exception thrown by Java APIs when a permission check fails. We may deprecate this API for removal at a later date, but for now the impact of doing so would be too high.
...
```

Reply:
Switched to this approach in 88c31c9f29f22754443e2c67b3e9e9e293972fcf